packages/osc-ui/src/components/TextInput/text-input.scss

```
@@ -0,0 +1,160 @@
+/* stylelint-disable no-descending-specificity */
+/* stylelint-disable selector-class-pattern */
+@use "../../styles/settings" as *;
+/* stylelint-disable plugin/selector-bem-pattern */
+
+$input-py: calc(var(--space-scale-micro) - 3px);
+$input-pr: var(--space-scale-delta);
+$input-pl: var(--space-scale-epsilon);
+
+.c-input {
+    display: flex;
+    order: 1;
+    padding: $input-py $input-pr $input-py $input-pl;
+    border: solid 1px var(--color-quinary);
+    font-size: var(--font-scale-base);
+    cursor: text;
+    grid-column: 1;
+    grid-row: 2;
+
+    &__icon {
+        justify-self: end;
+        align-self: center;
+        grid-row: 2;
+        grid-column: 1;
+        padding-right: 5%;
+    }
+
+    &:focus {
+        outline: 3px solid transparent;
+        box-shadow: 0 0 0 2px var(--color-primary);
+    }
+
+    &[disabled] {
+        background-color: var(--color-neutral-100);
+        color: var(--color-neutral-500);
+        pointer-events: none;
+    }
+
+    &__outer-container {
+        display: flex;
+    }
+
+    &__container {
+        display: grid;
+        width: 100%;
+
+        .c-label {
+            grid-row: 1;
+            grid-column: 1;
+        }
+
+        &--secondary {
+            .c-input {
+                border-radius: 6px;
+            }
+        }
+
+        &--tertiary,
+        &--quaternary {
+            position: relative;
+
+            .c-label {
+                display: flex;
+                align-items: center;
+                grid-row: 2;
+                transform-origin: top left;
+                transition: 100ms ease-in-out;
```

use variables for animation timing:
```suggestion
                transition: $timing-s ease-in-out;
```